### ISSUE TITLE
Conda activates a non-existent environment

### ISSUE THREAD

**Issue by chdoig** [conda/conda#3916]
```
(lux) cdoig@066-cdoig:~$ source activate sec-demo
(sec-demo) cdoig@066-cdoig:~$ conda list


CondaEnvironmentNotFoundError: Could not find environment: /Users/cdoig/anaconda/envs/sec-demo .
You can list all discoverable environments with `conda info --envs`.
```

Does conda caches environments somewhere else than the envs path?

conda info:
```
Current conda install:

               platform : osx-64
          conda version : 4.2.12
       conda is private : False
      conda-env version : 4.2.12
    conda-build version : 2.0.10
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : /Users/cdoig/anaconda  (writable)
    default environment : /Users/cdoig/anaconda/envs/sec-demo
       envs directories : /Users/cdoig/anaconda/envs
          package cache : /Users/cdoig/anaconda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /Users/cdoig/.condarc
           offline mode : False

```

```
(sec-demo) cdoig@066-cdoig:~$ cd /Users/cdoig/anaconda/envs/sec-demo
-bash: cd: /Users/cdoig/anaconda/envs/sec-demo: No such file or directory
```

**Comment by chdoig** [comment]
I figured out what is going on:

If I have a folder named, for example, research and I do: 

```
source activate research
```

Then, a bin folder gets created inside my research folder:

```
(research) cdoig@066-cdoig:~/research$ ll
total 16
drwxr-xr-x    7 cdoig  staff   238B Nov 18 10:32 ./
drwxr-xr-x+ 288 cdoig  staff   9.6K Nov 14 17:03 ../
-rw-r--r--@   1 cdoig  staff   6.0K Nov 18 10:33 .DS_Store
drwxr-xr-x    5 cdoig  staff   170B Nov 18 10:32 bin/
drwxr-xr-x   18 cdoig  staff   612B Dec 14  2014 my_first_gradle_project/
drwxr-xr-x    5 cdoig  staff   170B Dec 11  2014 test_gradle/
drwxr-xr-x   11 cdoig  staff   374B Jan 19  2015 topic_modelling/
```

with the following content:

```
(research) cdoig@066-cdoig:~/research/bin$ ls
activate@   conda@      deactivate@
```

I'm very confused with this behavior, @kalefranz

**Comment by nehaljwani** [comment]
A small workaround for this (if activating custom paths as conda envs is expected):
```bash
[wani@lenovo test]$ source /tmp/conda_install_dir/bin/activate
(root) [wani@lenovo test]$ mkdir throwawayenv
(root) [wani@lenovo test]$ source activate throwawayenv
(throwawayenv) [wani@lenovo test]$ tree throwawayenv/bin/
throwawayenv/bin/
├── activate -> /tmp/conda_install_dir/bin/activate
├── conda -> /tmp/conda_install_dir/bin/conda
└── deactivate -> /tmp/conda_install_dir/bin/deactivate

0 directories, 3 files
(throwawayenv) [wani@lenovo test]$ mkdir -p /tmp/conda_install_dir/envs
(throwawayenv) [wani@lenovo test]$ ln -s $PWD/throwawayenv/ /tmp/conda_install_dir/envs/
(throwawayenv) [wani@lenovo test]$ conda list
# packages in environment at /tmp/conda_install_dir/envs/throwawayenv:
#
(throwawayenv) [wani@lenovo test]$ source activate --help


CondaHelp: Usage: . activate [ENV] [-h] [-v]

Adds the 'bin' directory of the environment ENV to the front of PATH. ENV
may either refer to just the name of the environment, or the full prefix
path.

Where:
    ENV             the virtual environment to activate (dflt: root)
    -h,--help       shows this dialog
    -v,--verbose    shows more detailed info for the activate process
                    (useful when there are post-activate scripts)

Alternatively use the following variables when the above parameter passing
doesn't work:
    CONDA_ENVNAME="ENV"
    CONDA_HELP="true"
    CONDA_VERBOSE="true"

Example(s):
    . activate root
    CONDA_ENVNAME="root" ; . activate

```

**Comment by kalefranz** [comment]
@chdoig There are a few things going on here, all of it pretty wrong.

Technically, *any* directory, even an empty directory, is a conda environment.  At least that's been try in the past.  Except, not quite... `conda create` always forced you to specify at least one package.  That was dumb.  So in 4.2 we made `conda create` *actually* be able to create an empty env.  With the original spec of an empty env being just an empty directory, that's why your `research` directory is being activated I think.  An edge case we didn't have before.  I think it'll be pretty easy to fix though.  When we create an empty env, we actually touch a single file, at `CONDA_PREFIX/conda-meta/history`.  The existence of that file is the flag we can use to determine if a location is a conda env.

And... right now, when you activate a conda env, one of the many things that happens is that those three executables (`conda`, `activate`, and `deactivate`) get symlinked from the root env into the activated env.  I *despise* this behavior.  There's a ton of discussion about it.  We're doing something in 4.3 to change it.

**Comment by chdoig** [comment]
Thanks for the explanation @kalefranz!

**Comment by makmanalp** [comment]
Thanks for this! For the lazy future me / other people who run into this and don't want to read the whole thread, the solution is to symlink your local environment into the global envs/ directory in the anaconda install path.

The tricky behavior is that it looks like the environment correctly activated, which is technically true, because activation doesn't really check or do anything. Our use case is that we install conda environments into non-local shared directories because they get pretty huge, and often overflow the per-user space quotas we have.  Just better documenting "install an env to a specific path" would go a long way.

**Comment by kalefranz** [comment]
Fixed in conda 4.4.
